e2e/testsuite/tx.go

```diff
@@ -93,9 +91,16 @@ func containsMessage(s string, messages []string) bool {
 	return false
 }
 
-// AssertValidTxResponse verifies that an sdk.TxResponse
+// TODO: Maybe Allow specified error code too, see ensure ics20 transfer fails in base test (and there's one more)
+func (s *E2ETestSuite) AssertTxFailure(resp sdk.TxResponse, expectedMsg string) {
```

Comment:
I can see this argument list growing over time as we need to assert on different fields, what do you think of this construct


```go
type TxAssertion func(t *testing.T, resp sdk.TxResponse)


func AssertTxCode(code int) TxAssertion {
   return func(t *testing.T, tx sdk.TxResponse) {
	  require.Equal(t, code, int(tx.Code), "expected tx code to match")
   }
}

func AssertTxMessage(msg string) TxAssertion {
	return func(t *testing.T, tx sdk.TxResponse) {
		require.Contains(t, msg, string(tx.RawLog), "expected tx msg to match")
	}
}

func AssertTxSuccess(resp sdk.TxResponse) TxAssertion{
	return func(t *testing.T, resp sdk.TxResponse) {
		errorMsg := fmt.Sprintf("%+v", resp)
		s.Require().NotEmpty(resp.TxHash, errorMsg)
		s.Require().NotEqual(int64(0), resp.GasUsed, errorMsg)
		s.Require().NotEqual(int64(0), resp.GasWanted, errorMsg)
		s.Require().NotEmpty(resp.Events, errorMsg)
		s.Require().NotEmpty(resp.Data, errorMsg)
	}

}

func (s E2ETestSuite) AssertTx(tx sdk.TxResponse, assertions... TxAssertion) {
   for _, fn := range assertions {
      fn(s.T(), tx)
   }
}


func main(){
	s.AssertTx(tx, AssertTxCode(0), AssertTxMessage("foo"))
}
```

We could provide default assertion functions, to use in most tests, but also allows granular control for any assertion we want to make.

Just a thought!

Reply:
Yeah, I'm really liking this approach. Better to put them in a file of their own.

Reply:
cool! I think this sort of thing is actually a good usecase for `.` imports to help in readability so that we don't need to fully qualify the function each time. 

e.g. 

`assertions.AssertTxMessage("foo"), assertions.AssertTxMessage("foo"))`

is harder to read than

`AssertTxMessage("foo"), AssertTxMessage("foo"))`

unless we want to include the package name in the name.

i.e.

`assert.TxMessage("foo"), assert.TxMessage("foo"))`